src/installer/installer.go

```diff
@@ -115,19 +125,47 @@ func (i *installer) InstallNode() error {
 	} else {
 		i.log.Info("Done writing image to disk")
 	}
-	if isBootstrap {
-		i.UpdateHostInstallProgress(models.HostStageWaitingForControlPlane, "")
-		if err = errs.Wait(); err != nil {
-			i.log.Error(err)
+	return nil
+}
+
+func (i *installer) startInstallation(isSingleNodeBootstrap bool) error {
+	i.UpdateHostInstallProgress(models.HostStageStartingInstallation, i.Config.Role)
+
+	if err := i.ops.Mkdir(InstallDir); err != nil {
+		i.log.WithError(err).Errorf("Failed to create install dir")
+		return err
+	}
+
+	if !isSingleNodeBootstrap {
+		i.log.Infof("Start cleaning up device %s", i.Device)
+		err := i.cleanupInstallDevice()
+		if err != nil {
+			i.log.WithError(err).Errorf("failed to prepare install device %s", i.Device)
 			return err
 		}
+		i.log.Infof("Finished cleaning up device %s", i.Device)
+	}
+
+	return nil
+}
+
+func (i *installer) finalizingInstallation(isBootstrap bool, isSingleNodeBootstrap bool) error {
+
+	stage := models.HostStageRebooting
+	operation := i.ops.Reboot
+	if isSingleNodeBootstrap {
+		stage = models.HostStageDone
+		operation = i.ops.Shutdown
```

Comment:
I think it's safer to have no-op operation in case of single node

Reply:
possible, thought it will be better